src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.TranslationAction_Actions.cs

```diff
@@ -45,71 +51,18 @@ public override GeneratorDriver TransformGeneratorDriver(GeneratorDriver generat
 
             public override TranslationAction? TryMergeWithPrior(TranslationAction priorAction)
             {
-                if (priorAction is TouchDocumentAction priorTouchAction &&
-                    priorTouchAction._newState == _oldState)
+                if (priorAction is TouchDocumentsAction priorTouchAction &&
+                    priorTouchAction._newStates.SequenceEqual(_oldStates))
```

Comment:
>priorTouchAction._newStates.SequenceEqual(_oldStates) [](http://example.com/codeflow?start=20&length=53)

Is this more likely to happen than I think it is? (seems like a never would happen thing to me)

Reply:
it's very likely.  this is what happens when you get a stream of edits coming in for one document (basically, the  typing scenario).

we can either keep making a longer and longer chain of 'edits' where the compilation goes through N steps to get to hte end.  Or we can say that the actions 'merge' and we'll jump from teh starting compilation all the way to teh end step.